### PR TITLE
feat: add kibisis client

### PR DESCRIPTION
**Pull request by kieranroneill**
### Description

> _Please explain the changes you made here_
Adds a new client for the Kibisis wallet.

The new client takes advantage of the [`BroadcastChannel`](https://developer.mozilla.org/en-US/docs/Web/API/BroadcastChannel) API to exchange messages between the client (use-wallet) and the Kibisis web extension using a simple message bus as detailed below.

simple_message_bus

The client does not need any external SDKs or libraries and leverages standard Web APIs.

#### Kibisis Support

Kibisis will support `use-wallet` integration from version 1.10.0+.

#### Browser Compatibility

The `BroadcastChannel` API has been fully supported by all major browsers since March 2022, however, this only due to Safari finally releasing full support in version [15.4](https://developer.mozilla.org/en-US/docs/Web/API/BroadcastChannel#browser_compatibility) (2022-03-14).

### Checklist

> _Please, make sure to comply with the checklist below before expecting review_

- [x] Code compiles correctly
- [x] Created tests which fail without the change (if possible)
- [x] All tests passing
- [x] Extended the README / documentation, if necessary

### CODE REVIEW

**Comment by drichar** [comment]
Hi @kieranroneill 👋 Thank you for the PR! Kibisis will be a great addition to the library. Very nice work on the wallet... it's a clean UI with attention to detail. Congrats on the launch!

Please make sure CI checks pass and update the pull request's draft status when it's ready for review. Let me know if you have any other questions or if there's anything else I should know beforehand. Looking forward to getting this merged and released 👍

**Comment by kieranroneill** [comment]
> Hi @kieranroneill 👋 Thank you for the PR! Kibisis will be a great addition to the library. Very nice work on the wallet... it's a clean UI with attention to detail. Congrats on the launch!
> 
> Please make sure CI checks pass and update the pull request's draft status when it's ready for review. Let me know if you have any questions or if there's anything else I should know beforehand. Looking forward to getting this merged and released 👍

Thank you for your kind words. It will be a great milestone to have `use-wallet` integration. Your code is a key part of the AVM ecosystem and I am very impressed with your work; it is so good to have a connector!

I have just finished the wallet side of things and I am waiting until I get a release in the Chrome store and hopefully I can get this PR ready :smile:

**Comment by kieranroneill** [comment]
> Hi @kieranroneill 👋 Thank you for the PR! Kibisis will be a great addition to the library. Very nice work on the wallet... it's a clean UI with attention to detail. Congrats on the launch!
> 
> Please make sure CI checks pass and update the pull request's draft status when it's ready for review. Let me know if you have any questions or if there's anything else I should know beforehand. Looking forward to getting this merged and released 👍

Hey @drichar the Chrome Web Store has released the latest version, this should be good to go for review.

**Comment by drichar** [comment]
Excellent @kieranroneill! Thanks for the update, I will review this ASAP

File: src/clients/kibisis/client.ts [hunk 1]
```diff
@@ -0,0 +1,474 @@
...
+      const result = await KibisisClient.sendRequestWithTimeout<
+        GetProvidersParams,
+        GetProvidersResult
+      >({
+        method: 'getProviders',
+        params: {
+          providerId: ARC_0027_PROVIDER_ID
+        },
+        reference: ARC_0027_GET_PROVIDERS_REQUEST,
+        timeout: LOWER_REQUEST_TIMEOUT
```
[the remark this answers is not in the document]
Comment (kieranroneill) [reply src/clients/kibisis/client.ts:91]: > Is there a reason this timeout is so long? A full minute before timing out seems excessive, considering the `init` method is being called in a `Promise.all` initializing all of the configured providers.

You are absolutely right about this. The get providers should be fast and does not require any user input, so I will shorten it.

**Review by drichar — APPROVED** [review]
Thank you for addressing my comments. The 3s timeout for initialization is much better. And I was able to connect and sign the test transaction in Storybook using the latest version of the extension.

As we discussed (offline) I'll add a disclaimer in the documentation that Kibisis is in active development and currently only supports betanet and testnet, with a Discord link to report any issues.

I also think it makes more sense for the Storybook demo to connect to testnet, something else I'll change prior to release.

But I think this is ready to go, LGTM! Nice work @kieranroneill 👍

**Pull request merged**